presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -327,15 +334,42 @@ public StripeFooter readStripeFooter(StripeInformation stripe, AbstractAggregate
         }
     }
 
-    private Map<Integer, List<RowGroupIndex>> readColumnIndexes(Map<StreamId, Stream> streams, Map<StreamId, OrcInputStream> streamsData)
+    private Map<Integer, List<RowGroupBloomfilter>> readBloomfilterIndexes(Map<StreamId, Stream> streams, Map<StreamId, OrcInputStream> streamsData)
+            throws IOException
+    {
+        ImmutableMap.Builder<Integer, List<RowGroupBloomfilter>> bloomfilters = ImmutableMap.builder();
+        for (Entry<StreamId, Stream> entry : streams.entrySet()) {
+            Stream stream = entry.getValue();
+            if (stream.getStreamKind() == BLOOM_FILTER) {
+                OrcInputStream inputStream = streamsData.get(entry.getKey());
+                bloomfilters.put(stream.getColumn(), metadataReader.readBloomfilterIndexes(inputStream));
+            }
+        }
+        return bloomfilters.build();
+    }
+
+    private Map<Integer, List<RowGroupIndex>> readColumnIndexes(Map<StreamId, Stream> streams, Map<StreamId, OrcInputStream> streamsData, Map<Integer, List<RowGroupBloomfilter>> bloomfilterIndexes)
             throws IOException
     {
         ImmutableMap.Builder<Integer, List<RowGroupIndex>> columnIndexes = ImmutableMap.builder();
         for (Entry<StreamId, Stream> entry : streams.entrySet()) {
             Stream stream = entry.getValue();
             if (stream.getStreamKind() == ROW_INDEX) {
                 OrcInputStream inputStream = streamsData.get(entry.getKey());
-                columnIndexes.put(stream.getColumn(), metadataReader.readRowIndexes(inputStream));
+                List<RowGroupBloomfilter> bloomfilters = bloomfilterIndexes.get(stream.getColumn());
+                List<RowGroupIndex> rowGroupIndexes = metadataReader.readRowIndexes(inputStream);
+                if (bloomfilters != null && !bloomfilters.isEmpty()) {
+                    ImmutableList.Builder<RowGroupIndex> tmpRowGroupIndexes = ImmutableList.builder();
+                    int i = 0;
```

Comment:
I'd change this section to this which uses an esier to read for loop, a new `ColumnStatistics.withBloomFilter`, and assumes there is only one bloom filter:
```java
                    ImmutableList.Builder<RowGroupIndex> newRowGroupIndexes = ImmutableList.builder();
                    for (int i = 0; i < rowGroupIndexes.size(); i++) {
                        RowGroupIndex rowGroupIndex = rowGroupIndexes.get(i);
                        ColumnStatistics columnStatistics = rowGroupIndex.getColumnStatistics()
                                .withBloomFilter(bloomfilters.get(i));
                        newRowGroupIndexes.add(new RowGroupIndex(rowGroupIndex.getPositions(), columnStatistics));
                    }
                    rowGroupIndexes = newRowGroupIndexes.build();